src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitStorage = new HashMap<>();
+
+    public static void setBalance(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Quantity can't be negative");
+        }
+        if (fruitStorage.values().stream().anyMatch(value -> value < 0)) {
+            throw new RuntimeException("Balance can't be negative");
+        }
+        fruitStorage.put(fruit, quantity);
+    }
```

Comment:
The check for negative quantity is correct, but the check for negative balance in the storage is redundant here. The balance check should be made in the service responsible for handling the 'purchase' operation, not in the storage itself.